### PR TITLE
Allow re-deploy when current state is Published 

### PR DESCRIPTION
**Pull request by tmkasun**
image

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/Resources.jsx [hunk 1]
```diff
@@ -297,6 +297,12 @@ export default function Resources(props) {
             setOperationRateLimits(response.body.list);
         });
         // TODO: need to handle the error cases through catch ~tmkb
+        // if (api.isSOAPToREST()) {
+        //     api.getResourcePolicies().then((res) => {
+        //         const { a } = res;
+        //         console.log(a);
+        //     });
+        // }
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/Resources.jsx]: commented?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx [hunk 2]
```diff
@@ -171,29 +171,33 @@ class LifeCycleUpdate extends Component {
...
+            if (state.event === 'Deploy as a Prototype') {
+                return {
+                    ...state,
+                    disabled: !isPrototype || api.endpointConfig == null,
+                };
+            }
+            if (state.event === 'Publish') {
+                let { displayName } = state;
+                if (lcState.state === 'Published') {
+                    displayName = 'Re Deploy';
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx]:
```suggestion
                    displayName = 'Redeploy';
```

Comment (tmkasun) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7740/commits/cd50ac4ef19324d9f4ee4f3b6dc01cb590aeb3be

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/Resources.jsx (hunk 1 above)
Comment (tmkasun) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/Resources.jsx]: Removed in https://github.com/wso2/carbon-apimgt/pull/7740/commits/cd50ac4ef19324d9f4ee4f3b6dc01cb590aeb3be

**Pull request merged**